src/vs/workbench/api/node/extHostTerminalService.ts

```diff
@@ -23,6 +23,16 @@ export class ExtHostTerminalService extends BaseExtHostTerminalService {
 	public createTerminalFromOptions(options: vscode.TerminalOptions, internalOptions?: ITerminalInternalOptions): vscode.Terminal {
 		const terminal = new ExtHostTerminal(this._proxy, generateUuid(), options, options.name);
 		this._terminals.push(terminal);
+		if (options.location && typeof options.location === 'object' && 'parentTerminal' in options.location) {
+			const parentTerminal = options.location.parentTerminal;
+			if (parentTerminal) {
+				const parentExtHostTerminal = this._terminals.find(t => t.value === parentTerminal);
+				internalOptions = internalOptions ? internalOptions : {};
+				if (parentExtHostTerminal) {
+					internalOptions.parentTerminal = parentExtHostTerminal;
```

Comment:
I think it's confusing/overkill to include location in `ITerminalInternalOptions`, since we only need to do that for this case. I'll add a comment in the `ITerminalInternalOptions` interface explaining that we need to resolve the `ExtHostTerminal `from the `Terminal `provided